mne/annotations.py

```diff
@@ -0,0 +1,79 @@
+# Authors: Jaakko Leppakangas <jaeilepp@student.jyu.fi>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from datetime import datetime
+import time
+import json
+
+
+class Annotations(object):
+    """Annotation object for annotating segments of raw data.
+
+    Parameters
+    ----------
+    onset: array of float, shape (n_annotations,)
+        Annotation time onsets from the beginning of the recording.
+    duration: array of float, shape (n_annotations,)
+        Durations of the annotations.
+    description: array of str, shape (n_annotations,)
+        Array of strings containing description for each annotation.
+    orig_time: float | int | instance of datetime | array of int | None
+        A POSIX Timestamp, datetime or an array containing the timestamp as the
+        first element and microseconds as the second element. Determines the
+        starting time of annotation acquisition. If None (default),
+        starting time is determined from beginning of raw data.
+    """
+
+    def __init__(self, onset, duration, description, orig_time=None):
+
+        if orig_time is not None:
+            if isinstance(orig_time, datetime):
+                orig_time = float(time.mktime(orig_time.timetuple()))
+            elif not np.isscalar(orig_time):
+                orig_time = orig_time[0] + orig_time[1] / 1000000.
+            else:  # isscalar
+                orig_time = float(orig_time)  # np.int not serializable
+        self.orig_time = orig_time
+
+        onset = np.array(onset)
+        if onset.ndim != 1:
+            raise ValueError('Onset must be a one dimensional array.')
+        duration = np.array(duration)
+        if duration.ndim != 1:
+            raise ValueError('Duration must be a one dimensional array.')
+        if not (len(onset) == len(duration) == len(description)):
+            raise ValueError('Onset, duration and description must be '
+                             'equal in sizes.')
+        # sort the segments by start time
+        order = onset.argsort(axis=0)
+        self.onset = onset[order]
+        self.duration = duration[order]
+        self.description = np.array(description)[order]
+
+    def _serialize(self):
+        """Function that serializes the annotation object for saving."""
+        return json.dumps({'onset': self.onset.tolist(),
+                           'duration': self.duration.tolist(),
+                           'description': self.description.tolist(),
+                           'orig_time': self.orig_time})
```

Comment:
I am -1 on string serialization. I don't think it's the most compatible option. I think we should either use FIFF constants and tags properly, or use HDF5 instead of inventing this third hybrid way of IO.

For the FIFF route, instead of onset/duration, we can just use tmin/tmax (even though they're mis-named a bit for this purpose), and we already have time and description, so I think you can use the more standard way of writing FIFF as:

```
FIFF.FIFF_MEAS_DATE   = 204
FIFF.FIFF_COMMENT     = 206
FIFF.FIFF_MNE_BASELINE_MIN           = 3568     # Time of baseline beginning
FIFF.FIFF_MNE_BASELINE_MAX           = 3569     # Time of baseline end
```
This way, other tools (MNE-C, MNE-MATLAB, etc.) won't be forced to implement JSON parsing to read data. It should also be more compact byte-wise to store it using binary format. The JSON dump feels a bit like a hack to me, but maybe I'm alone in that...

Alternatively we can switch to HDF5, which should totally future-compatible, give us compression for free, etc. But then it's harder for other packages to read. So I think I'd prefer using FIFF constants properly instead of making use of this new JSON-dump technique.